oak-lucene/src/main/java/org/apache/jackrabbit/oak/plugins/index/lucene/writer/IndexWriterUtils.java

```diff
@@ -70,6 +73,10 @@ public static IndexWriterConfig getIndexWriterConfig(LuceneIndexDefinition defin
             IndexWriterConfig config = new IndexWriterConfig(VERSION, analyzer);
             if (remoteDir) {
                 config.setMergeScheduler(new SerialMergeScheduler());
+            } else {
```

Comment:
one reason why remoteDir could be true is because copyOnWrite is disabled (which usually isn't the case), which would imply we would most probably run into the else block here even during normal indexing.

So effectively the new code change here (the else block) will impact normal indexing as well (not just the out of the band indexing from oak-run), from what I can tell - was that taken into account ?

Reply:
I am not sure. Can you suggest what change would be advisable here to ccount for that. The default is 1 thread so essentially the concurrent will default to serial ofcourse if the property is not overriden

Reply:
> The default is 1 thread so essentially the concurrent will default to serial ofcourse if the property is not overriden

How's the default 1 ? The max threads are hardcoded to 8 as of now - or did I miss something ?

Reply:
As you suggested the properties should be configurable and if that's the case then the default should be 1. But question is what can be done if in future this default changes?

Reply:
ah - ok. I think it would be difficult to determine here whether this is being called from out-of-band indexing or redular async indexing. So I think the best we can do is to set the default to 1 and add a comment that the default value should be changed only after thorough analysis.